Send SIGHUP to node with grps enabled process crash

**Tendermint version** (use `tendermint version` or `git rev-parse --verify HEAD` if installed from source): 0.34.14

**ABCI app** (name for built-in, URL for self-written if it's publicly available): none

**Environment**:
- **OS** (e.g. from /etc/os-release): debian 11 

**How to reproduce it** (as minimally and precisely as possible): kill -SIGTERM  process_id

**Logs (paste a small part showing an error (< 10 lines) or link a pastebin, gist, etc. containing more of the log file)**:
```
10:35PM INF WSJSONRPC method=subscribe module=rpc-server protocol=websocket remote={"IP":"192.168.90.31","Port":59376,"Zone":""}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa10d80]

goroutine 3913950 [running]:
github.com/tendermint/tendermint/libs/pubsub.(*Subscription).Out(0xc00171d558)
        /home/com/go/pkg/mod/github.com/tendermint/tendermint@v0.34.14/libs/pubsub/subscription.go:43
github.com/tendermint/tendermint/rpc/core.Subscribe.func1()
        /home/com/go/pkg/mod/github.com/tendermint/tendermint@v0.34.14/rpc/core/events.go:50 +0x9e
created by github.com/tendermint/tendermint/rpc/core.Subscribe
        /home/com/go/pkg/mod/github.com/tendermint/tendermint@v0.34.14/rpc/core/events.go:47 +0x485
```

**node command runtime flags**: node start

--- a/CHANGELOG_PENDING.md
+++ b/CHANGELOG_PENDING.md
@@ -10,7 +10,7 @@ Friendly reminder, we have a [bug bounty program](https://hackerone.com/tendermi
 
 - CLI/RPC/Config
 
-  - [config] \#7230 rpc: Add experimental config params to allow for subscription buffer size control (@thanethomson).
+  - [config] [\#7230](https://github.com/tendermint/tendermint/issues/7230) rpc: Add experimental config params to allow for subscription buffer size control (@thanethomson).
 
 - Apps
 
@@ -26,5 +26,6 @@ Friendly reminder, we have a [bug bounty program](https://hackerone.com/tendermi
 
 ### BUG FIXES
 
+- [\#7309](https://github.com/tendermint/tendermint/issues/7309) pubsub: Report a non-nil error when shutting down (fixes #7306).
 - [\#7057](https://github.com/tendermint/tendermint/pull/7057) Import Postgres driver support for the psql indexer (@creachadair).
 - [\#7106](https://github.com/tendermint/tendermint/pull/7106) Revert mutex change to ABCI Clients (@tychoish).

--- a/libs/pubsub/pubsub.go
+++ b/libs/pubsub/pubsub.go
@@ -194,7 +194,7 @@ func (s *Server) subscribe(ctx context.Context, clientID string, query Query, ou
 	case <-ctx.Done():
 		return nil, ctx.Err()
 	case <-s.Quit():
-		return nil, nil
+		return nil, errors.New("service is shutting down")
 	}
 }
 

